### PR TITLE
Add attr_accessor for data_uri and fallback for filename

### PR DESCRIPTION
**Pull request by kntmrkm**
Hi I posted this issue. then sorry about my misunderstood.
https://github.com/timsly/carrierwave-data-uri/issues/3

BTW, I added two implementation.
- attr_accessor for #{column}_data_uri # if without this I should put every model which use image upload.
- filename fallback # if without filename. set random filename.

Could you please accept this?
thanks.

### CODE REVIEW

File: lib/carrierwave_data_uri/tempfile.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ class Tempfile < ::Tempfile
       attr_writer :original_filename
 
       def original_filename
-        @original_filename || File.basename(@tmpname)
+        @original_filename
```
Comment (timfjord) [inline lib/carrierwave_data_uri/tempfile.rb:9]: why `"temp.#{extension}"` is better than `File.basename(@tmpname)`?

File: lib/carrierwave_data_uri/mount.rb [hunk 2]
```diff
@@ -8,10 +8,15 @@ def mount_uploader(column, uploader=nil, options={}, &block)
...
 
         class_eval <<-RUBY, __FILE__, __LINE__+1
+          attr_accessor :#{column}_data_uri
           attr_accessor :#{column}_data_filename
 
           def #{column}_data_uri=(data)
-            self.#{column} = Parser.new(data).to_file original_filename: self.#{column}_data_filename
+            original_filename = self.#{column}_data_filename.presence
+
+            if data.present?
```
Comment (timfjord) [inline lib/carrierwave_data_uri/mount.rb:17]: `Parse` raises an exception on bad data, what is point to check it here?

File: lib/carrierwave_data_uri/tempfile.rb (hunk 1 above)
Comment (kntmrkm) [inline lib/carrierwave_data_uri/tempfile.rb:9]: I want to use `image` file upload with form post.

not below as README.

```
user = User.find 123
user.avatar_data_filename = 'somefile.jpg'
user.avatar_data_uri = 'data:image/gif;base64,R0lGODlhAQABAJEAAAAAAP////8AAP///yH5BAEAAAMALAAAAAABAAEAAAICVAEAOw=='
user.save
```

With form post, there is no `filename`. so, add `temp filename`.

File: lib/carrierwave_data_uri/mount.rb (hunk 2 above)
Comment (kntmrkm) [inline lib/carrierwave_data_uri/mount.rb]: Yes. As you said.
There is no need check presence of data.

Comment (kntmrkm) [inline lib/carrierwave_data_uri/mount.rb:17]: This is same to https://github.com/timsly/carrierwave-data-uri/pull/6/files#r54526876.

If use file upload with form post.
There are empty `image_data_uri` attributes.
then, to skip upload process.

Comment (timfjord) [inline lib/carrierwave_data_uri/mount.rb:17]: @kntmrkm You need control this in your app

File: lib/carrierwave_data_uri/tempfile.rb (hunk 1 above)
Comment (timfjord) [inline lib/carrierwave_data_uri/tempfile.rb:9]: You can control this in the app

**Comment by kntmrkm** [comment]
yes, as you said. I can control those points.
But, It will be dirty code with form post. so, I changed like this PR.

thank you for your feedback.

File: lib/carrierwave_data_uri/mount.rb (hunk 2 above)
Comment (timfjord) [inline lib/carrierwave_data_uri/mount.rb:17]: @kntmrkm i am thinking to add something like silent mode. If you enable this mode than parser won't raise an exception on bad data. So stay tuned

Comment (kntmrkm) [inline lib/carrierwave_data_uri/mount.rb:17]: @timsly OK. Thanks. Looking forward it!

**Pull request closed without merging**